airflow/www/forms.py

```diff
@@ -234,13 +234,42 @@ def process(self, formdata=None, obj=None, **kwargs):
                 "corresponding Airflow Provider Package."
             ),
         )
-        description = StringField(lazy_gettext("Description"), widget=BS3TextAreaFieldWidget())
-        host = StringField(lazy_gettext("Host"), widget=BS3TextFieldWidget())
-        schema = StringField(lazy_gettext("Schema"), widget=BS3TextFieldWidget())
-        login = StringField(lazy_gettext("Login"), widget=BS3TextFieldWidget())
-        password = PasswordField(lazy_gettext("Password"), widget=BS3PasswordFieldWidget())
-        port = IntegerField(lazy_gettext("Port"), validators=[Optional()], widget=BS3TextFieldWidget())
-        extra = TextAreaField(lazy_gettext("Extra"), widget=BS3TextAreaFieldWidget())
+
+        description = StringField(
+            lazy_gettext("Description"),
+            widget=BS3TextAreaFieldWidget(),
+            validators=[NoHTMLTags()],
+        )
+        host = StringField(
+            lazy_gettext("Host"),
+            widget=BS3TextFieldWidget(),
+            validators=[NoHTMLTags()],
+        )
+        schema = StringField(
+            lazy_gettext("Schema"),
+            widget=BS3TextFieldWidget(),
+            validators=[NoHTMLTags()],
+        )
+        login = StringField(
+            lazy_gettext("Login"),
+            widget=BS3TextFieldWidget(),
+            validators=[NoHTMLTags()],
+        )
+        password = PasswordField(
+            lazy_gettext("Password"),
+            widget=BS3PasswordFieldWidget(),
+            validators=[NoHTMLTags()],
+        )
+        port = IntegerField(
+            lazy_gettext("Port"),
+            validators=[Optional()],
+            widget=BS3TextFieldWidget(),
+        )
+        extra = TextAreaField(
```

Comment:
Just checked the issue and found:
>  Also when they do, they should be escaped when rendered on HTML pages.

Also, I'm wondering if some users need to pass HTML via extra; should we block that completely or just stop rendering the HTML by escaping them on the connection web UI?